ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/attestation/ValidateableAttestation.java

```diff
@@ -50,15 +56,37 @@ private ValidateableAttestation(
     this.hashTreeRoot = Suppliers.memoize(attestation::hash_tree_root);
   }
 
+  public Optional<IndexedAttestation> getMaybeIndexedAttestation() {
+    return maybeIndexedAttestation;
+  }
+
   public IndexedAttestation getIndexedAttestation() {
     return maybeIndexedAttestation.orElseThrow(
         () ->
             new UnsupportedOperationException(
                 "ValidateableAttestation does not have an IndexedAttestation yet."));
   }
 
-  public void setIndexedAttestation(IndexedAttestation maybeIndexedAttestation) {
-    this.maybeIndexedAttestation = Optional.of(maybeIndexedAttestation);
+  public Optional<Bytes32> getMaybeCommitteeShufflingSeed() {
+    return maybeCommitteeShufflingSeed;
+  }
+
+  public Bytes32 getCommitteeShufflingSeed() {
+    return maybeCommitteeShufflingSeed.orElseThrow(
+        () ->
+            new UnsupportedOperationException(
+                "ValidateableAttestation does not have a committee shuffling seed yet."));
+  }
+
+  public void setIndexedAttestation(IndexedAttestation indexedAttestation) {
+    this.maybeIndexedAttestation = Optional.of(indexedAttestation);
+  }
+
+  public void saveCommitteeShufflingSeed(BeaconState state) {
+    Bytes32 committeeShufflingSeed =
+        get_seed(
+            state, compute_epoch_at_slot(attestation.getData().getSlot()), DOMAIN_BEACON_ATTESTER);
```

Comment:
I think we should simplify this further to just store the randao rather than spending time calculating the entire shuffling seed.  If the randao matches, the shuffling seed will as well.  So just:

`get_randao_mix(state, compute_epoch_at_slot(attestation.getData().getSlot()))`

Reply:
Sorry, this is wrong - we need to do the same randaoIndex calculation as in `get_seed` so worth extracting that bit of `get_seed` as a method we can reuse but it effectively becomes:
```suggestion
    UInt64 epoch = compute_epoch_at_slot(attestation.getData().getSlot());
    UInt64 randaoIndex = epoch.plus(EPOCHS_PER_HISTORICAL_VECTOR - MIN_SEED_LOOKAHEAD - 1);
    return get_randao_mix(state, randaoIndex);
```

Reply:
Following conversation with Proto, sticking with seed probably does make sense - just make sure that we calculate the seed for the block only once, not once per attestation or group.

Reply:
Makes sense to me.

Reply:
Done.